mesonbuild/modules/rust.py

```diff
@@ -250,6 +254,15 @@ def bindgen(self, state: ModuleState, args: T.List, kwargs: FuncBindgen) -> Modu
 
         if self._bindgen_bin is None:
             self._bindgen_bin = state.find_program('bindgen', wanted=kwargs['bindgen_version'])
+            if self._bindgen_rust_target is not None:
+                # ExternalCommand.command's type is bonkers
+                _, _, err = mesonlib.Popen_safe(
+                    T.cast('T.List[str]', self._bindgen_bin.get_command()) +
+                    ['--rust-target', self._bindgen_rust_target])
+                # Sometimes this is "invalid Rust target" and sometimes "invalid
+                # rust target"
+                if 'Got an invalid' in err:
+                    self._bindgen_rust_target = None
```

Comment:
https://github.com/rust-lang/rust-bindgen/commit/7c569c99e4b63656d1308610913f4bed8a817ef6

New check should look something like

```
Rust/5.0 (Glibc; Linux x86_64; rv:135.0) rustc/20100101 Rustup/1.84.1
```

Reply:
> With the changes done in this PR, it would be possible to pass any* Rust version to --rust-target and bindgen will automatically enable the Rust features that are compatible with that version. In our example, that means that it would be possible to use --rust-version=1.41.1 explicitly, or even better, read the value from rust-version so they don't have to manually change it in the future.

So, we can just check the version of bindgen and given new enough bindgen, I guess, we can stop bothering to check if it's accepted, as it always will be?

Reply:
Should bindgen (for >=0.71.0) also be getting `--rust-edition` if `rust_std` is set in meson project options?

https://github.com/rust-lang/rust-bindgen/blob/20aa65a0b9edfd5f8ab3e038197da5cb2c52ff18/CHANGELOG.md?plain=1#L241